src/libraries/System.Private.CoreLib/ref/System.Private.CoreLib.ExtraApis.cs

```diff
@@ -37,3 +37,13 @@ public static partial class Debug
         public static System.Diagnostics.DebugProvider SetProvider(System.Diagnostics.DebugProvider provider) { throw null; }
     }
 }
+
+#if FEATURE_WASM_THREADS
+namespace System.Runtime.InteropServices.JavaScript
+{
+    public class JSProxyContextBase {
```

Comment:
I would prefer to implement this is just 

src\mono\System.Private.CoreLib\src\System\Threading\Monitor.Mono.cs
```
namespace System.Threading
{
    public static partial class Monitor
    {
        [ThreadStatic]
        public static bool ThrowOnBlockingWait;
    }
}
```

Which we would set just once and we would not have to subclass `JSProxyContext`